src/types/redis_bloom_chain.cc

```diff
@@ -185,14 +178,35 @@ rocksdb::Status BloomChain::Exist(const Slice &user_key, const Slice &item, int
 
   std::string item_string = item.ToString();
   // check
+  bool exist = false;
   for (int i = metadata.n_filters - 1; i >= 0; --i) {  // TODO: to test which direction for searching is better
-    s = bloomCheck(bf_key_list[i], item_string, ret);
+    s = bloomCheck(bf_key_list[i], item_string, &exist);
     if (!s.ok()) return s;
-    if (*ret == 1) {
+    if (exist) {
+      *ret = 1;
       break;
     }
   }
 
   return rocksdb::Status::OK();
 }
+
+rocksdb::Status BloomChain::Info(const Slice &user_key, BloomFilterInfo *info) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+
+  BloomChainMetadata metadata;
+  rocksdb::Status s = getBloomChainMetadata(ns_key, &metadata);
+  if (s.IsNotFound()) return rocksdb::Status::NotFound("key is not found");
```

Comment:
Why not simply use `if (!s.ok()) return s;` that will return the `rocksdb::Status::NotFound` status as well? Then check on the caller side if the status is NotFound if necessary.

Reply:
Sure, It's a mistake left by solution 1. I will fix it.